app/src/forms/form/service.js

```diff
@@ -31,6 +31,14 @@ const service = {
 
       await Form.query(trx).insert(obj);
       if (data.identityProviders && Array.isArray(data.identityProviders) && data.identityProviders.length) {
+        const fips = [];
+        for (const p of data.identityProviders) {
+          const exists = await IdentityProvider.query(trx).where('code', p.code).where('active', true).first();
+          if (!exists) {
+            throw new Problem(422, `${p.code} is not a valid Identity Provider code`);
+          }
+          fips.push({id: uuidv4(), formId: obj.id, code: p.code, createdBy: currentUser.username});
```

Comment:
The fips array - is it used anywhere else outside of the loop or the if block? Not seeing it immediately anywhere else in immediate code - could drop the const declaration and push declarations if they're unused.

Reply:
ha, nice one. yeah, it's supposed to be used in the insert...fixed.